python/cudf/cudf/tests/test_dtypes.py

```diff
@@ -34,3 +34,35 @@ def test_cdf_to_pandas(data, ordered):
         pd.CategoricalDtype(data, ordered)
         == cudf.CategoricalDtype(categories=data, ordered=ordered).to_pandas()
     )
+
+
+@pytest.mark.parametrize(
+    "data", [{"a": [[1, 2, 3]]}, pd.DataFrame({"a": [[1, 2, 3]]})]
```

Comment:
Could we add more tests here to check that we raise the same error for the below cases for example?

* Empty list?
* More than one list
* Combination of list and numeric data?

Ditto below in DataFrame.